tests/platform_tests/sfp/test_sfputil.py

```diff
@@ -40,6 +208,97 @@ def check_interfaces_up(duthost, namespace, up_ports):
         return False
 
 
+def get_namespace_cmd_option(duthost, asic_index):
+    """Get the namespace option used in the command"""
+    namespace = duthost.get_namespace_from_asic_id(asic_index)
+    return "-n {}".format(namespace) if namespace else ""
+
+
+def get_down_ports(duthost, ports):
+    """Check and return the down ports among the given ports."""
+    return duthost.show_interface(command="status", up_ports=ports)["ansible_facts"][
+        "ansible_interface_link_down_ports"]
+
+
+def is_interface_status_expected(duthost, ports, expect_up=True):
+    """Check if the given ports are up or down as expected."""
+    if expect_up:
+        return len(get_down_ports(duthost, ports)) == 0
+    else:
+        return len(get_down_ports(duthost, ports)) == len(ports)
+
+
+def check_interface_status(duthost, ports, expect_up=True, wait_time=None):
+    """
+    Check if the given ports are up or down as expected.
+
+    Args:
+        duthost: DUT host object
+        ports: List of ports to check
+        expect_up: True if the ports are expected to be up, False if down
+        wait_time: Time to wait for the ports to come up or down
+    """
+    expect_status_str = "up" if expect_up else "down"
+    err_msg = ""
+
+    if wait_time is None:
+        port_down_wait_time, port_up_wait_time = \
+            default_port_toggle_wait_time(duthost, len(ports))
+        if expect_up:
+            wait_time = port_up_wait_time
+        else:
+            wait_time = port_down_wait_time
+
+    logging.info("Wait for ports to come {}: {}".format(expect_status_str, ports))
+    is_ok = wait_until(wait_time, 5, 0,
```

Comment:
@longhuan-cisco Can we change the polling interval from 5s to 1s to optimize test case run time?

Reply:
updated